src/DistributedLocking/Consul/LockProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Cascade\DistributedLocking\Consul;
+
+use Cascade\DistributedLocking\ILockProvider;
+use Cascade\DistributedLocking\ILockSessionProvider;
+use SensioLabs\Consul\Services\KV;
+
+class LockProvider implements ILockProvider
+{
+    /** @var KV */
+    private $consulKeyValue;
+
+    /** @var ILockSessionProvider */
+    private $sessionProvider;
+
+    /**
+     * @param ILockSessionProvider $sessionProvider
+     * @param KV $consulKeyValue
+     */
+    public function __construct(ILockSessionProvider $sessionProvider, KV $consulKeyValue)
```

Comment:
Also is the `ILockSessionProvider` actually interchangeable for other session providers? Or does it need to be a `Cascade\DistributedLocking\Consul\LockSessionProvider` specifically for this to work?